### PR TITLE
94911 run cost-allocation as deployment

### PR DESCRIPTION
**Pull request by nilsgstrabo**
- Use a cron module in code to schedule collection and storage of Prometheus data
- Removed the CronJob chart template
- Added a Deployment chart template

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,63 +3,106 @@ package main
...
+		if err := moveResourceRequestsFromPrometheusToSQLDB(promClient, sqlClient); err != nil {
+			log.Error(err)
+		}
+	})
+
+	c.Run()
 }
 
-func printCostBetweenDates(from, to time.Time, promClient PrometheusClient, sqlClient SQLClient) {
+func printCostBetweenDates(from, to time.Time, promClient clients.PrometheusClient, sqlClient clients.SQLClient) error {
```
Comment (satr) [inline main.go:68]: Not used?

File: main.go [hunk 2]
```diff
@@ -3,63 +3,106 @@ package main
...
+	return nil
 }
 
-func moveResourceRequestsFromPrometheusToSQLDB(promClient PrometheusClient, sqlClient SQLClient) {
+func moveResourceRequestsFromPrometheusToSQLDB(promClient clients.PrometheusClient, sqlClient clients.SQLClient) error {
 	measuredTimeUTC := time.Now().UTC()
 	reqResources, err := promClient.GetRequiredResources(measuredTimeUTC)
 	if err != nil {
-		log.Fatal("Error getting required resources: ", err.Error())
+		return errors.WithMessage(err, "eror getting required resources")
```
Comment (satr) [inline main.go:88]: Typo "eror"

**Review by satr — CHANGES_REQUESTED** [review]
Some remarks

File: main.go (hunk 1 above)
Comment (nilsgstrabo) [reply main.go:68]: Probably just for debugging purposes. Removed.

File: README.md [hunk 3]
```diff
@@ -24,4 +24,13 @@ Run following command to set tag (with corresponding version)
...
 git push origin v1.0.0
+```
+
+## Debugging locally
+
+Create a copy of .env.template and name it .env. Set variables to allow local debugging. This file is ignored by git.
+
+Prometheus operator must be port forwarded to a local port:
+```
+k port-forward pod/prometheus-prometheus-operator-prometheus-0 9090:9090
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply README.md:35]: fixed

**Review by satr — APPROVED** [review]

**Pull request merged**